src/ringct/rctSigs.cpp

```diff
@@ -717,7 +770,10 @@ namespace rct {
         CHECK_AND_ASSERT_MES(rv.type == RCTTypeFull, false, "verRct called on non-full rctSig");
         if (semantics)
         {
-          CHECK_AND_ASSERT_MES(rv.outPk.size() == rv.p.rangeSigs.size(), false, "Mismatched sizes of outPk and rv.p.rangeSigs");
+          if (rv.p.rangeSigs.empty())
```

Comment:
I'm not a big fan of this way of determining the type of range proofs by relying on the Borromean sigs being empty, because it doesn't check for other erroneous cases such as both Borromean and BP sigs being non-empty. Probably this is unimportant, and I also don't know what would be a better way.